HAWKULAR-450

mwringe opened the pull request:
changes to make standalone charting work with Hawkular Metrics directly.

mwringe committed:
HAWKULAR-450: changes to make standalone charting work with Hawkular Metrics directly.

mtho11 commented:
@mwringe can you move those changes over to the Typescript file (the javascript file is generated from the typescript file): 
https://github.com/hawkular/hawkular-charts/blob/master/src/chart/hawkular-chart-directive.ts

The js file will get clobbered the next time any of the*.ts files change. You can type 'gulp watch' and it will watch your files and recompile anything that changes. Or just 'gulp' will build the js file from the ts files.

mwringe committed:
HAWKULAR-450: Update the ts file instead of the generated js one, update the generated js file to the lastest.

mwringe commented:
Ok, updated the ts file and updated the hawkular-metrics.js file to the latest generated one.

There are a lot of changes in the latest generated hawkular-metrics.js, mostly due to white space issues but also some code changes.

The whitespace issue may be due to some configuration on my machine, if there is a particular step I need to do for the whitespace to be generated differently let me know.

The code differences (other than what is new in the ts file) maybe due to the generated hawkular-metrics.js file being a bit out of sync with the latest from the ts file?

mtho11 commented on src/chart/hawkular-chart-directive.ts:
```diff
@@ -446,11 +447,11 @@ module Charts {
             if (response) {
-
               return response.map((point:IChartDataPoint) => {
+                var timestamp = point.start + (point.start - point.end)/2;
```
I have some concern with how this will affect some chart types as they will be shifted to the right a bit because of using a midpoint timestamp instead of start timestamp.
This only affects stand alone charts (not normal charts that supply their own data), so its not a big deal.

mtho11 replied:
Maybe rename timestamp to midPointTimestamp to be more clear.

mwringe replied:
There is no point.timestamp with bucketed data, so before it would just render nothing at all.

You may need to handle bucketed data differently and use different logic when dealing with bucketed data (eg midpoint calculating) and when using non-bucketed data (eg just use point.timestamp since that data is available)

mtho11 replied:
Ah, good to know. This is one of those things that changed a while back and another reason why the standalone charts weren't working. Thanks!

LGTM.

mtho11 replied:
Why not just use start instead of (point.start - point.end)/2  ?

mwringe replied:
The data being returned is for the time between the start and the end values. Which is why I think it makes sense to use the midpoint between them.

Eg say the bucket is for the timespan between 10:00 and 10:30, I would think putting the dot at 10:15 makes more sense than putting it at 10:00

But looking at the code, I made a huge mistake here. It should be:

point.start + (point.end - point.start)/2

instead of:

point.start + (point.start - point.end)/2

mtho11 replied:
Yes, I agree with that.
